graal-js/src/com.oracle.truffle.js/src/com/oracle/truffle/js/runtime/JSRealm.java

```
@@ -1983,16 +1988,48 @@ private JSDynamicObject createIntlObject() {
         JSObjectUtil.putDataProperty(context, graalObject, Strings.LANGUAGE, Strings.fromJavaString(JavaScriptLanguage.NAME), flags);
         assert GRAALVM_VERSION != null;
         JSObjectUtil.putDataProperty(context, graalObject, Strings.VERSION_GRAAL_VM, GRAALVM_VERSION, flags);
         JSObjectUtil.putDataProperty(context, graalObject, Strings.VERSION_ECMA_SCRIPT, esVersion, flags);
         JSObjectUtil.putDataProperty(context, graalObject, Strings.IS_GRAAL_RUNTIME, JSFunction.create(this, isGraalRuntimeFunction(context)), flags);
+        if (options.getUnhandledRejectionsMode() == JSContextOptions.UnhandledRejectionsTrackingMode.HANDLER) {
+            JSFunctionObject registerFunction = JSFunction.create(this, registerPromiseRejectionHandlerFunction());
+            JSObjectUtil.putDataProperty(context, graalObject, Strings.REGISTER_PROMISE_REJECTION_HANDLER, registerFunction, flags);
+        }
         putGlobalProperty(Strings.GRAAL, graalObject);
     }
 
+    private JSFunctionData registerPromiseRejectionHandlerFunction() {
+        return context.getOrCreateBuiltinFunctionData(BuiltinFunctionKey.RegisterPromiseRejectionHandler, (c) -> {
+            return JSFunctionData.createCallOnly(c, new JavaScriptRootNode(c.getLanguage(), null, null) {
+                @Override
+                public Object execute(VirtualFrame frame) {
+                    Object[] args = frame.getArguments();
+                    if (JSArguments.getUserArgumentCount(args) > 0) {
+                        Object arg = JSArguments.getUserArgument(args, 0);
+                        if (JSRuntime.isNullOrUndefined(arg)) {
+                            registerHandler(null);
+                        } else {
+                            registerHandler(arg);
+                        }
+                    } else {
+                        registerHandler(null);
+                    }
+                    return Undefined.instance;
```

I think that `undefined` should be allowed (to disable the handler programmatically), but this code allows one to register any arbitrary (non-executable) handler, e.g.:
```
> Graal.registerPromiseRejectionHandler(42)
> Promise.reject('foo')
TypeError: execute on 42 failed due to: Message not supported.
```
we should probably just throw an error here if the handler is not executable (nor `undefined/null`).